appstudio-controller/controllers/appstudio.redhat.com/deploymenttargetbinder_controller.go

```diff
@@ -85,30 +85,57 @@ func (r *DeploymentTargetClaimReconciler) Reconcile(ctx context.Context, req ctr
 	// Handle deletion if the DTC has a deletion timestamp set.
 	if dtc.GetDeletionTimestamp() != nil {
 		log.Info("Handling a deleted DeploymentTargetClaim")
-		// If the DTC is bound set the status of the corresponding DT to Released
+		// If the DTC is bound set, handle the corresponding DT
 		if isDTCBound(dtc) {
 			dt, err := getDTBoundByDTC(ctx, r.Client, &dtc)
 			if err != nil && !apierr.IsNotFound(err) {
 				return ctrl.Result{}, err
 			}
 
 			if dt != nil {
-				err = updateDTStatusPhase(ctx, r.Client, dt, applicationv1alpha1.DeploymentTargetPhase_Released, log)
-				if err != nil {
+				var dtcls *applicationv1alpha1.DeploymentTargetClass
+				if dtcls, err = findMatchingDTClassForDT(ctx, r.Client, *dt); err != nil {
 					return ctrl.Result{}, err
 				}
-				log.Info("DeploymentTarget is released", "DeploymentTarget", dt.Name)
+				// Add the deletion finalizer if it is absent.
+				if addFinalizer(dt, FinalizerDT) {
+					if err = r.Client.Update(ctx, dt); err != nil {
+						return ctrl.Result{}, fmt.Errorf("failed to add finalizer %s to DeploymentTarget %s in namespace %s: %v", FinalizerDT, dt.Name, dt.Namespace, err)
+					}
+					log.Info("Added finalizer to DeploymentTarget", "finalizer", FinalizerDT)
+				}
+
+				if dtcls.Spec.ReclaimPolicy == applicationv1alpha1.ReclaimPolicy_Delete {
+					err = r.Client.Delete(ctx, dt)
+					if err != nil {
+						return ctrl.Result{}, err
+					}
+					log.Info("DeploymentTarget is marked to Deleted", "DeploymentTarget", dt.Name)
+					return ctrl.Result{}, nil
+					//} else if dtcls.Spec.ReclaimPolicy == applicationv1alpha1.ReclaimPolicy_Retain {
```

Comment:
I think we can probably remove this comment for now - if additional reclaim policies get added, that can be handled in a follow-up ticket in the future.

Reply:
OK. Thanks!